### PR TITLE
don't upsert job->bc flow when deleteing bc

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1549448

@openshift/sig-developer-experience ptal

I'll create a new extended test to cover the delete verification scenario once this is in the openshift jenkins centos image.

### CODE REVIEW

**Comment by gabemontero** [comment]
[test]

**Comment by gabemontero** [comment]
@bparees - independent of the bugzilla (delete on the openshift side getting no-op'ed), I added in a separate commit that filters deleting the bc *when the job is deleted* if the sync plugin did not create the bc as well

From our various side discussions, I *think* you want this behavior as well, but of course chime in as needed.

Also, @hrishin @rupalibehera - you all on the OSIO / fabric side might be interested in these changes ... they are related to recent commits from your end pulled into this repo

File: src/main/java/io/fabric8/jenkins/openshiftsync/Annotations.java [hunk 1]
```diff
@@ -20,5 +20,6 @@
 public class Annotations {
 	public static final String JENKINS_JOB_PATH = "jenkins.openshift.io/job-path";
 	public static final String GENERATED_BY = "jenkins.openshift.io/generated-by";
+    public static final String GENERATED_BY_JENKINS = "jenkins";
```
Comment (bparees) [inline src/main/java/io/fabric8/jenkins/openshiftsync/Annotations.java]: indenting

File: src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java [hunk 2]
```diff
@@ -161,8 +178,11 @@ private void upsertItemGroup(ItemGroup itemGroup) {
 
 	private void upsertWorkflowJob(WorkflowJob job) {
 		BuildConfigProjectProperty property = buildConfigProjectForJob(job);
-		if (property != null) {
-			logger.info("Upsert WorkflowJob " + job.getName() + " to BuildConfig: " + property.getNamespace() + "/"
+		if (property != null && 
+		    (!BuildConfigWatcher.isDeleteInProgress(
+		            property.getNamespace()+property.getName()))) {
+			logger.info("Upsert WorkflowJob " + job.getName() +
```
Comment (bparees) [inline src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java]: indenting?

File: src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java [hunk 3]
```diff
@@ -119,9 +122,23 @@ public void onDeleted(Item item) {
 				BuildConfig buildConfig = getAuthenticatedOpenShiftClient().buildConfigs().inNamespace(namespace)
 						.withName(buildConfigName).get();
 				if (buildConfig != null) {
+				    boolean generatedBySyncPlugin = false;
+				    Map<String,String> annotations = buildConfig.getMetadata().getAnnotations();
+				    if (annotations != null) {
+				        generatedBySyncPlugin = Annotations.GENERATED_BY_JENKINS.equals(annotations.get(Annotations.GENERATED_BY));
+				    }
 					try {
```
Comment (bparees) [inline src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java]: some more weird indenting

File: src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java (hunk 2 above)
Comment (gabemontero) [reply src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java]: its for format with 80 chars a line thing

File: src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java (hunk 3 above)
Comment (gabemontero) [reply src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java]: its for format with 80 chars a line thing

**Comment by gabemontero** [comment]
fyi ... leaving commits separate this time ... address orthogonal/unrelated items

**Comment by gabemontero** [comment]
[merge]

**Pull request merged**